// @subframe/core/dist/cjs/components/radix/context-menu.cjs
"use client";
'use strict';

var RadixContextMenu = require('@radix-ui/react-context-menu');

function _interopNamespaceDefault(e) {
	var n = Object.create(null);
	if (e) {
		Object.keys(e).forEach(function (k) {
			if (k !== 'default') {
				var d = Object.getOwnPropertyDescriptor(e, k);
				Object.defineProperty(n, k, d.get ? d : {
					enumerable: true,
					get: function () { return e[k]; }
				});
			}
		});
	}
	n.default = e;
	return Object.freeze(n);
}

var RadixContextMenu__namespace = /*#__PURE__*/_interopNamespaceDefault(RadixContextMenu);

var Arrow = RadixContextMenu__namespace.Arrow;
var CheckboxItem = RadixContextMenu__namespace.CheckboxItem;
var Content = RadixContextMenu__namespace.Content;
var ContextMenu = RadixContextMenu__namespace.ContextMenu;
var ContextMenuArrow = RadixContextMenu__namespace.ContextMenuArrow;
var ContextMenuCheckboxItem = RadixContextMenu__namespace.ContextMenuCheckboxItem;
var ContextMenuContent = RadixContextMenu__namespace.ContextMenuContent;
var ContextMenuGroup = RadixContextMenu__namespace.ContextMenuGroup;
var ContextMenuItem = RadixContextMenu__namespace.ContextMenuItem;
var ContextMenuItemIndicator = RadixContextMenu__namespace.ContextMenuItemIndicator;
var ContextMenuLabel = RadixContextMenu__namespace.ContextMenuLabel;
var ContextMenuPortal = RadixContextMenu__namespace.ContextMenuPortal;
var ContextMenuRadioGroup = RadixContextMenu__namespace.ContextMenuRadioGroup;
var ContextMenuRadioItem = RadixContextMenu__namespace.ContextMenuRadioItem;
var ContextMenuSeparator = RadixContextMenu__namespace.ContextMenuSeparator;
var ContextMenuSub = RadixContextMenu__namespace.ContextMenuSub;
var ContextMenuSubContent = RadixContextMenu__namespace.ContextMenuSubContent;
var ContextMenuSubTrigger = RadixContextMenu__namespace.ContextMenuSubTrigger;
var ContextMenuTrigger = RadixContextMenu__namespace.ContextMenuTrigger;
var Group = RadixContextMenu__namespace.Group;
var Item = RadixContextMenu__namespace.Item;
var ItemIndicator = RadixContextMenu__namespace.ItemIndicator;
var Label = RadixContextMenu__namespace.Label;
var Portal = RadixContextMenu__namespace.Portal;
var RadioGroup = RadixContextMenu__namespace.RadioGroup;
var RadioItem = RadixContextMenu__namespace.RadioItem;
var Root = RadixContextMenu__namespace.Root;
var Separator = RadixContextMenu__namespace.Separator;
var Sub = RadixContextMenu__namespace.Sub;
var SubContent = RadixContextMenu__namespace.SubContent;
var SubTrigger = RadixContextMenu__namespace.SubTrigger;
var Trigger = RadixContextMenu__namespace.Trigger;

exports.Arrow = Arrow;
exports.CheckboxItem = CheckboxItem;
exports.Content = Content;
exports.ContextMenu = ContextMenu;
exports.ContextMenuArrow = ContextMenuArrow;
exports.ContextMenuCheckboxItem = ContextMenuCheckboxItem;
exports.ContextMenuContent = ContextMenuContent;
exports.ContextMenuGroup = ContextMenuGroup;
exports.ContextMenuItem = ContextMenuItem;
exports.ContextMenuItemIndicator = ContextMenuItemIndicator;
exports.ContextMenuLabel = ContextMenuLabel;
exports.ContextMenuPortal = ContextMenuPortal;
exports.ContextMenuRadioGroup = ContextMenuRadioGroup;
exports.ContextMenuRadioItem = ContextMenuRadioItem;
exports.ContextMenuSeparator = ContextMenuSeparator;
exports.ContextMenuSub = ContextMenuSub;
exports.ContextMenuSubContent = ContextMenuSubContent;
exports.ContextMenuSubTrigger = ContextMenuSubTrigger;
exports.ContextMenuTrigger = ContextMenuTrigger;
exports.Group = Group;
exports.Item = Item;
exports.ItemIndicator = ItemIndicator;
exports.Label = Label;
exports.Portal = Portal;
exports.RadioGroup = RadioGroup;
exports.RadioItem = RadioItem;
exports.Root = Root;
exports.Separator = Separator;
exports.Sub = Sub;
exports.SubContent = SubContent;
exports.SubTrigger = SubTrigger;
exports.Trigger = Trigger;
