// @subframe/core/dist/cjs/components/radix/dropdown-menu.cjs
"use client";
'use strict';

var RadixDropdownMenu = require('@radix-ui/react-dropdown-menu');

function _interopNamespaceDefault(e) {
	var n = Object.create(null);
	if (e) {
		Object.keys(e).forEach(function (k) {
			if (k !== 'default') {
				var d = Object.getOwnPropertyDescriptor(e, k);
				Object.defineProperty(n, k, d.get ? d : {
					enumerable: true,
					get: function () { return e[k]; }
				});
			}
		});
	}
	n.default = e;
	return Object.freeze(n);
}

var RadixDropdownMenu__namespace = /*#__PURE__*/_interopNamespaceDefault(RadixDropdownMenu);

var Arrow = RadixDropdownMenu__namespace.Arrow;
var CheckboxItem = RadixDropdownMenu__namespace.CheckboxItem;
var Content = RadixDropdownMenu__namespace.Content;
var DropdownMenu = RadixDropdownMenu__namespace.DropdownMenu;
var DropdownMenuArrow = RadixDropdownMenu__namespace.DropdownMenuArrow;
var DropdownMenuCheckboxItem = RadixDropdownMenu__namespace.DropdownMenuCheckboxItem;
var DropdownMenuContent = RadixDropdownMenu__namespace.DropdownMenuContent;
var DropdownMenuGroup = RadixDropdownMenu__namespace.DropdownMenuGroup;
var DropdownMenuItem = RadixDropdownMenu__namespace.DropdownMenuItem;
var DropdownMenuItemIndicator = RadixDropdownMenu__namespace.DropdownMenuItemIndicator;
var DropdownMenuLabel = RadixDropdownMenu__namespace.DropdownMenuLabel;
var DropdownMenuPortal = RadixDropdownMenu__namespace.DropdownMenuPortal;
var DropdownMenuRadioGroup = RadixDropdownMenu__namespace.DropdownMenuRadioGroup;
var DropdownMenuRadioItem = RadixDropdownMenu__namespace.DropdownMenuRadioItem;
var DropdownMenuSeparator = RadixDropdownMenu__namespace.DropdownMenuSeparator;
var DropdownMenuSub = RadixDropdownMenu__namespace.DropdownMenuSub;
var DropdownMenuSubContent = RadixDropdownMenu__namespace.DropdownMenuSubContent;
var DropdownMenuSubTrigger = RadixDropdownMenu__namespace.DropdownMenuSubTrigger;
var DropdownMenuTrigger = RadixDropdownMenu__namespace.DropdownMenuTrigger;
var Group = RadixDropdownMenu__namespace.Group;
var Item = RadixDropdownMenu__namespace.Item;
var ItemIndicator = RadixDropdownMenu__namespace.ItemIndicator;
var Label = RadixDropdownMenu__namespace.Label;
var Portal = RadixDropdownMenu__namespace.Portal;
var RadioGroup = RadixDropdownMenu__namespace.RadioGroup;
var RadioItem = RadixDropdownMenu__namespace.RadioItem;
var Root = RadixDropdownMenu__namespace.Root;
var Separator = RadixDropdownMenu__namespace.Separator;
var Sub = RadixDropdownMenu__namespace.Sub;
var SubContent = RadixDropdownMenu__namespace.SubContent;
var SubTrigger = RadixDropdownMenu__namespace.SubTrigger;
var Trigger = RadixDropdownMenu__namespace.Trigger;

exports.Arrow = Arrow;
exports.CheckboxItem = CheckboxItem;
exports.Content = Content;
exports.DropdownMenu = DropdownMenu;
exports.DropdownMenuArrow = DropdownMenuArrow;
exports.DropdownMenuCheckboxItem = DropdownMenuCheckboxItem;
exports.DropdownMenuContent = DropdownMenuContent;
exports.DropdownMenuGroup = DropdownMenuGroup;
exports.DropdownMenuItem = DropdownMenuItem;
exports.DropdownMenuItemIndicator = DropdownMenuItemIndicator;
exports.DropdownMenuLabel = DropdownMenuLabel;
exports.DropdownMenuPortal = DropdownMenuPortal;
exports.DropdownMenuRadioGroup = DropdownMenuRadioGroup;
exports.DropdownMenuRadioItem = DropdownMenuRadioItem;
exports.DropdownMenuSeparator = DropdownMenuSeparator;
exports.DropdownMenuSub = DropdownMenuSub;
exports.DropdownMenuSubContent = DropdownMenuSubContent;
exports.DropdownMenuSubTrigger = DropdownMenuSubTrigger;
exports.DropdownMenuTrigger = DropdownMenuTrigger;
exports.Group = Group;
exports.Item = Item;
exports.ItemIndicator = ItemIndicator;
exports.Label = Label;
exports.Portal = Portal;
exports.RadioGroup = RadioGroup;
exports.RadioItem = RadioItem;
exports.Root = Root;
exports.Separator = Separator;
exports.Sub = Sub;
exports.SubContent = SubContent;
exports.SubTrigger = SubTrigger;
exports.Trigger = Trigger;
