// @subframe/core/dist/cjs/components/radix/hover-card.cjs
"use client";
'use strict';

var RadixHoverCard = require('@radix-ui/react-hover-card');

function _interopNamespaceDefault(e) {
	var n = Object.create(null);
	if (e) {
		Object.keys(e).forEach(function (k) {
			if (k !== 'default') {
				var d = Object.getOwnPropertyDescriptor(e, k);
				Object.defineProperty(n, k, d.get ? d : {
					enumerable: true,
					get: function () { return e[k]; }
				});
			}
		});
	}
	n.default = e;
	return Object.freeze(n);
}

var RadixHoverCard__namespace = /*#__PURE__*/_interopNamespaceDefault(RadixHoverCard);

var Arrow = RadixHoverCard__namespace.Arrow;
var Content = RadixHoverCard__namespace.Content;
var HoverCard = RadixHoverCard__namespace.HoverCard;
var HoverCardArrow = RadixHoverCard__namespace.HoverCardArrow;
var HoverCardContent = RadixHoverCard__namespace.HoverCardContent;
var HoverCardPortal = RadixHoverCard__namespace.HoverCardPortal;
var HoverCardTrigger = RadixHoverCard__namespace.HoverCardTrigger;
var Portal = RadixHoverCard__namespace.Portal;
var Root = RadixHoverCard__namespace.Root;
var Trigger = RadixHoverCard__namespace.Trigger;

exports.Arrow = Arrow;
exports.Content = Content;
exports.HoverCard = HoverCard;
exports.HoverCardArrow = HoverCardArrow;
exports.HoverCardContent = HoverCardContent;
exports.HoverCardPortal = HoverCardPortal;
exports.HoverCardTrigger = HoverCardTrigger;
exports.Portal = Portal;
exports.Root = Root;
exports.Trigger = Trigger;
