// @subframe/core/dist/cjs/components/radix/popover.cjs
"use client";
'use strict';

var RadixPopover = require('@radix-ui/react-popover');

function _interopNamespaceDefault(e) {
	var n = Object.create(null);
	if (e) {
		Object.keys(e).forEach(function (k) {
			if (k !== 'default') {
				var d = Object.getOwnPropertyDescriptor(e, k);
				Object.defineProperty(n, k, d.get ? d : {
					enumerable: true,
					get: function () { return e[k]; }
				});
			}
		});
	}
	n.default = e;
	return Object.freeze(n);
}

var RadixPopover__namespace = /*#__PURE__*/_interopNamespaceDefault(RadixPopover);

var Anchor = RadixPopover__namespace.Anchor;
var Arrow = RadixPopover__namespace.Arrow;
var Close = RadixPopover__namespace.Close;
var Content = RadixPopover__namespace.Content;
var Popover = RadixPopover__namespace.Popover;
var PopoverAnchor = RadixPopover__namespace.PopoverAnchor;
var PopoverArrow = RadixPopover__namespace.PopoverArrow;
var PopoverClose = RadixPopover__namespace.PopoverClose;
var PopoverContent = RadixPopover__namespace.PopoverContent;
var PopoverPortal = RadixPopover__namespace.PopoverPortal;
var PopoverTrigger = RadixPopover__namespace.PopoverTrigger;
var Portal = RadixPopover__namespace.Portal;
var Root = RadixPopover__namespace.Root;
var Trigger = RadixPopover__namespace.Trigger;

exports.Anchor = Anchor;
exports.Arrow = Arrow;
exports.Close = Close;
exports.Content = Content;
exports.Popover = Popover;
exports.PopoverAnchor = PopoverAnchor;
exports.PopoverArrow = PopoverArrow;
exports.PopoverClose = PopoverClose;
exports.PopoverContent = PopoverContent;
exports.PopoverPortal = PopoverPortal;
exports.PopoverTrigger = PopoverTrigger;
exports.Portal = Portal;
exports.Root = Root;
exports.Trigger = Trigger;
