// @subframe/core/dist/cjs/components/radix/radio-group.cjs
"use client";
'use strict';

var RadixRadioGroup = require('@radix-ui/react-radio-group');

function _interopNamespaceDefault(e) {
	var n = Object.create(null);
	if (e) {
		Object.keys(e).forEach(function (k) {
			if (k !== 'default') {
				var d = Object.getOwnPropertyDescriptor(e, k);
				Object.defineProperty(n, k, d.get ? d : {
					enumerable: true,
					get: function () { return e[k]; }
				});
			}
		});
	}
	n.default = e;
	return Object.freeze(n);
}

var RadixRadioGroup__namespace = /*#__PURE__*/_interopNamespaceDefault(RadixRadioGroup);

var Root = RadixRadioGroup__namespace.Root;
var Item = RadixRadioGroup__namespace.Item;
var Indicator = RadixRadioGroup__namespace.Indicator;
var RadioGroup = RadixRadioGroup__namespace.RadioGroup;
var RadioGroupItem = RadixRadioGroup__namespace.RadioGroupItem;
var RadioGroupIndicator = RadixRadioGroup__namespace.RadioGroupIndicator;

exports.Indicator = Indicator;
exports.Item = Item;
exports.RadioGroup = RadioGroup;
exports.RadioGroupIndicator = RadioGroupIndicator;
exports.RadioGroupItem = RadioGroupItem;
exports.Root = Root;
