// @subframe/core/dist/cjs/components/radix/tooltip.cjs
"use client";
'use strict';

var RadixTooltip = require('@radix-ui/react-tooltip');

function _interopNamespaceDefault(e) {
	var n = Object.create(null);
	if (e) {
		Object.keys(e).forEach(function (k) {
			if (k !== 'default') {
				var d = Object.getOwnPropertyDescriptor(e, k);
				Object.defineProperty(n, k, d.get ? d : {
					enumerable: true,
					get: function () { return e[k]; }
				});
			}
		});
	}
	n.default = e;
	return Object.freeze(n);
}

var RadixTooltip__namespace = /*#__PURE__*/_interopNamespaceDefault(RadixTooltip);

var Root = RadixTooltip__namespace.Root;
var Tooltip = RadixTooltip__namespace.Tooltip;
var TooltipArrow = RadixTooltip__namespace.TooltipArrow;
var TooltipContent = RadixTooltip__namespace.TooltipContent;
var TooltipPortal = RadixTooltip__namespace.TooltipPortal;
var TooltipProvider = RadixTooltip__namespace.TooltipProvider;
var TooltipTrigger = RadixTooltip__namespace.TooltipTrigger;
var Arrow = RadixTooltip__namespace.Arrow;
var Content = RadixTooltip__namespace.Content;
var Portal = RadixTooltip__namespace.Portal;
var Provider = RadixTooltip__namespace.Provider;
var Trigger = RadixTooltip__namespace.Trigger;

exports.Arrow = Arrow;
exports.Content = Content;
exports.Portal = Portal;
exports.Provider = Provider;
exports.Root = Root;
exports.Tooltip = Tooltip;
exports.TooltipArrow = TooltipArrow;
exports.TooltipContent = TooltipContent;
exports.TooltipPortal = TooltipPortal;
exports.TooltipProvider = TooltipProvider;
exports.TooltipTrigger = TooltipTrigger;
exports.Trigger = Trigger;
