// @subframe/core/dist/esm/components/radix/context-menu.mjs
"use client";
import * as RadixContextMenu from '@radix-ui/react-context-menu';

var Arrow = RadixContextMenu.Arrow;
var CheckboxItem = RadixContextMenu.CheckboxItem;
var Content = RadixContextMenu.Content;
var ContextMenu = RadixContextMenu.ContextMenu;
var ContextMenuArrow = RadixContextMenu.ContextMenuArrow;
var ContextMenuCheckboxItem = RadixContextMenu.ContextMenuCheckboxItem;
var ContextMenuContent = RadixContextMenu.ContextMenuContent;
var ContextMenuGroup = RadixContextMenu.ContextMenuGroup;
var ContextMenuItem = RadixContextMenu.ContextMenuItem;
var ContextMenuItemIndicator = RadixContextMenu.ContextMenuItemIndicator;
var ContextMenuLabel = RadixContextMenu.ContextMenuLabel;
var ContextMenuPortal = RadixContextMenu.ContextMenuPortal;
var ContextMenuRadioGroup = RadixContextMenu.ContextMenuRadioGroup;
var ContextMenuRadioItem = RadixContextMenu.ContextMenuRadioItem;
var ContextMenuSeparator = RadixContextMenu.ContextMenuSeparator;
var ContextMenuSub = RadixContextMenu.ContextMenuSub;
var ContextMenuSubContent = RadixContextMenu.ContextMenuSubContent;
var ContextMenuSubTrigger = RadixContextMenu.ContextMenuSubTrigger;
var ContextMenuTrigger = RadixContextMenu.ContextMenuTrigger;
var Group = RadixContextMenu.Group;
var Item = RadixContextMenu.Item;
var ItemIndicator = RadixContextMenu.ItemIndicator;
var Label = RadixContextMenu.Label;
var Portal = RadixContextMenu.Portal;
var RadioGroup = RadixContextMenu.RadioGroup;
var RadioItem = RadixContextMenu.RadioItem;
var Root = RadixContextMenu.Root;
var Separator = RadixContextMenu.Separator;
var Sub = RadixContextMenu.Sub;
var SubContent = RadixContextMenu.SubContent;
var SubTrigger = RadixContextMenu.SubTrigger;
var Trigger = RadixContextMenu.Trigger;

export { Arrow, CheckboxItem, Content, ContextMenu, ContextMenuArrow, ContextMenuCheckboxItem, ContextMenuContent, ContextMenuGroup, ContextMenuItem, ContextMenuItemIndicator, ContextMenuLabel, ContextMenuPortal, ContextMenuRadioGroup, ContextMenuRadioItem, ContextMenuSeparator, ContextMenuSub, ContextMenuSubContent, ContextMenuSubTrigger, ContextMenuTrigger, Group, Item, ItemIndicator, Label, Portal, RadioGroup, RadioItem, Root, Separator, Sub, SubContent, SubTrigger, Trigger };
