// @subframe/core/dist/cjs/packages/subframe-core/external/react-day-picker/dist/esm/labels/index.cjs
'use strict';

var labelGrid = require('./labelGrid.cjs');
var labelGridcell = require('./labelGridcell.cjs');
var labelDayButton = require('./labelDayButton.cjs');
var labelNav = require('./labelNav.cjs');
var labelMonthDropdown = require('./labelMonthDropdown.cjs');
var labelNext = require('./labelNext.cjs');
var labelPrevious = require('./labelPrevious.cjs');
var labelWeekday = require('./labelWeekday.cjs');
var labelWeekNumber = require('./labelWeekNumber.cjs');
var labelWeekNumberHeader = require('./labelWeekNumberHeader.cjs');
var labelYearDropdown = require('./labelYearDropdown.cjs');



exports.labelCaption = labelGrid.labelCaption;
exports.labelGrid = labelGrid.labelGrid;
exports.labelGridcell = labelGridcell.labelGridcell;
exports.labelDay = labelDayButton.labelDay;
exports.labelDayButton = labelDayButton.labelDayButton;
exports.labelNav = labelNav.labelNav;
exports.labelMonthDropdown = labelMonthDropdown.labelMonthDropdown;
exports.labelNext = labelNext.labelNext;
exports.labelPrevious = labelPrevious.labelPrevious;
exports.labelWeekday = labelWeekday.labelWeekday;
exports.labelWeekNumber = labelWeekNumber.labelWeekNumber;
exports.labelWeekNumberHeader = labelWeekNumberHeader.labelWeekNumberHeader;
exports.labelYearDropdown = labelYearDropdown.labelYearDropdown;
