// @subframe/core/dist/cjs/components/collapsible.cjs
"use client";
'use strict';

var _tslib = require('../_virtual/_tslib.cjs');
var RadixCollapsible = require('@radix-ui/react-collapsible');
var reactSlot = require('@radix-ui/react-slot');
var classNames = require('classnames');
var React = require('react');
var collapsible_module = require('./collapsible.module.css.cjs');

function _interopNamespaceDefault(e) {
  var n = Object.create(null);
  if (e) {
    Object.keys(e).forEach(function (k) {
      if (k !== 'default') {
        var d = Object.getOwnPropertyDescriptor(e, k);
        Object.defineProperty(n, k, d.get ? d : {
          enumerable: true,
          get: function () { return e[k]; }
        });
      }
    });
  }
  n.default = e;
  return Object.freeze(n);
}

var RadixCollapsible__namespace = /*#__PURE__*/_interopNamespaceDefault(RadixCollapsible);

var Chevron = React.forwardRef(function CollapsibleChevron(_a, ref) {
    var className = _a.className, otherProps = _tslib.__rest(_a, ["className"]);
    return React.createElement(reactSlot.Slot, _tslib.__assign({ ref: ref, className: classNames(className, collapsible_module.default.chevron) }, otherProps));
});
var Trigger = React.forwardRef(function Trigger(_a, ref) {
    var className = _a.className, children = _a.children, otherProps = _tslib.__rest(_a, ["className", "children"]);
    return (React.createElement(RadixCollapsible__namespace.Trigger, _tslib.__assign({ className: classNames(className, collapsible_module.default.trigger), ref: ref }, otherProps), children));
});
var Content = React.forwardRef(function Content(_a, ref) {
    var className = _a.className, children = _a.children, otherProps = _tslib.__rest(_a, ["className", "children"]);
    return (React.createElement(RadixCollapsible__namespace.Content, _tslib.__assign({ className: classNames(className, collapsible_module.default.content), ref: ref }, otherProps), children));
});
var Root = RadixCollapsible__namespace.Root;
var Collapsible = {
    Root: Root,
    Trigger: Trigger,
    Content: Content,
    Chevron: Chevron,
};

exports.Chevron = Chevron;
exports.Collapsible = Collapsible;
exports.Content = Content;
exports.Root = Root;
exports.Trigger = Trigger;
