// @subframe/core/dist/cjs/components/slider.cjs
"use client";
'use strict';

var _tslib = require('../_virtual/_tslib.cjs');
var RadixSlider = require('@radix-ui/react-slider');
var classNames = require('classnames');
var React = require('react');
var slider_module = require('./slider.module.css.cjs');

function _interopNamespaceDefault(e) {
  var n = Object.create(null);
  if (e) {
    Object.keys(e).forEach(function (k) {
      if (k !== 'default') {
        var d = Object.getOwnPropertyDescriptor(e, k);
        Object.defineProperty(n, k, d.get ? d : {
          enumerable: true,
          get: function () { return e[k]; }
        });
      }
    });
  }
  n.default = e;
  return Object.freeze(n);
}

var RadixSlider__namespace = /*#__PURE__*/_interopNamespaceDefault(RadixSlider);

var Root = React.forwardRef(function ProgressRoot(_a, ref) {
    var className = _a.className, children = _a.children, _b = _a.defaultValue, defaultValue = _b === void 0 ? [27] : _b, otherProps = _tslib.__rest(_a, ["className", "children", "defaultValue"]);
    return (React.createElement(RadixSlider__namespace.Root, _tslib.__assign({ ref: ref, className: classNames(className, slider_module.default.sliderRoot), defaultValue: defaultValue }, otherProps), children));
});
var Track = React.forwardRef(function Indicator(_a, ref) {
    var className = _a.className, children = _a.children, otherProps = _tslib.__rest(_a, ["className", "children"]);
    return (React.createElement(RadixSlider__namespace.Track, _tslib.__assign({ className: classNames(className, slider_module.default.sliderTrack), ref: ref }, otherProps), children));
});
var Range = React.forwardRef(function Indicator(_a, ref) {
    var className = _a.className, otherProps = _tslib.__rest(_a, ["className"]);
    return React.createElement(RadixSlider__namespace.Range, _tslib.__assign({ className: classNames(className, slider_module.default.sliderRange), ref: ref }, otherProps));
});
var Thumb = React.forwardRef(function Indicator(_a, ref) {
    var className = _a.className, otherProps = _tslib.__rest(_a, ["className"]);
    return React.createElement(RadixSlider__namespace.Thumb, _tslib.__assign({ className: classNames(className, slider_module.default.sliderThumb), ref: ref }, otherProps));
});
var Slider = { Root: Root, Track: Track, Range: Range, Thumb: Thumb };

exports.Range = Range;
exports.Root = Root;
exports.Slider = Slider;
exports.Thumb = Thumb;
exports.Track = Track;
