// @subframe/core/dist/esm/components/collapsible.mjs
"use client";
import { __rest, __assign } from '../_virtual/_tslib.mjs';
import * as RadixCollapsible from '@radix-ui/react-collapsible';
import { Slot } from '@radix-ui/react-slot';
import classNames from 'classnames';
import React__default from 'react';
import styles from './collapsible.module.css.mjs';

var Chevron = React__default.forwardRef(function CollapsibleChevron(_a, ref) {
    var className = _a.className, otherProps = __rest(_a, ["className"]);
    return React__default.createElement(Slot, __assign({ ref: ref, className: classNames(className, styles.chevron) }, otherProps));
});
var Trigger = React__default.forwardRef(function Trigger(_a, ref) {
    var className = _a.className, children = _a.children, otherProps = __rest(_a, ["className", "children"]);
    return (React__default.createElement(RadixCollapsible.Trigger, __assign({ className: classNames(className, styles.trigger), ref: ref }, otherProps), children));
});
var Content = React__default.forwardRef(function Content(_a, ref) {
    var className = _a.className, children = _a.children, otherProps = __rest(_a, ["className", "children"]);
    return (React__default.createElement(RadixCollapsible.Content, __assign({ className: classNames(className, styles.content), ref: ref }, otherProps), children));
});
var Root = RadixCollapsible.Root;
var Collapsible = {
    Root: Root,
    Trigger: Trigger,
    Content: Content,
    Chevron: Chevron,
};

export { Chevron, Collapsible, Content, Root, Trigger };
