// @subframe/core/dist/esm/components/slider.mjs
"use client";
import { __rest, __assign } from '../_virtual/_tslib.mjs';
import * as RadixSlider from '@radix-ui/react-slider';
import classNames from 'classnames';
import React__default from 'react';
import styles from './slider.module.css.mjs';

var Root = React__default.forwardRef(function ProgressRoot(_a, ref) {
    var className = _a.className, children = _a.children, _b = _a.defaultValue, defaultValue = _b === void 0 ? [27] : _b, otherProps = __rest(_a, ["className", "children", "defaultValue"]);
    return (React__default.createElement(RadixSlider.Root, __assign({ ref: ref, className: classNames(className, styles.sliderRoot), defaultValue: defaultValue }, otherProps), children));
});
var Track = React__default.forwardRef(function Indicator(_a, ref) {
    var className = _a.className, children = _a.children, otherProps = __rest(_a, ["className", "children"]);
    return (React__default.createElement(RadixSlider.Track, __assign({ className: classNames(className, styles.sliderTrack), ref: ref }, otherProps), children));
});
var Range = React__default.forwardRef(function Indicator(_a, ref) {
    var className = _a.className, otherProps = __rest(_a, ["className"]);
    return React__default.createElement(RadixSlider.Range, __assign({ className: classNames(className, styles.sliderRange), ref: ref }, otherProps));
});
var Thumb = React__default.forwardRef(function Indicator(_a, ref) {
    var className = _a.className, otherProps = __rest(_a, ["className"]);
    return React__default.createElement(RadixSlider.Thumb, __assign({ className: classNames(className, styles.sliderThumb), ref: ref }, otherProps));
});
var Slider = { Root: Root, Track: Track, Range: Range, Thumb: Thumb };

export { Range, Root, Slider, Thumb, Track };
